### ISSUE TITLE
options(warn = 1) by default?

### ISSUE THREAD

**Issue by HenrikBengtsson** [markvanderloo/tinytest#74]
WRE now (https://github.com/wch/r-source/commit/3a3073b94eaadb024e4ebf6cdfb041833bad4382) says:

```
@item
Tests should use @code{options(warn = 1)} as reporting

@example
There were 22 warnings (use warnings() to see them)
@end example

@noindent
is pointless, especially for automated checking systems.
```

Is this something that **tinytest** could do by default, e.g. in `run_test_file()`?

PS. This is just a comment; I'm new to **tinytest**, so sorry if this has already been discussed or is out of the scope of the design/philosophy.

**Comment by markvanderloo** [comment]
Thanks, I was unaware of this WRE requirement.

It seems like a good default, especially for `test_package` as this is the command that is intended to be used to trigger tests in  `R CMD check`.

**Comment by markvanderloo** [comment]
There are some other things I should set, e.g.  make sure that the sorting locale is `C` like in R CMD check. cf #69. In the works.

**Comment by eddelbuettel** [comment]
Interesting suggestion!  

But I am not sure if _always_ turning it on is right?   Maybe making this is a toggle is better.  I.e. in my for-work package, I warn on use of a deprecated-but-supported interface.  That counts as a warning _yet I know full well it is there_.  Making the test fails does not help.  

Or maybe we add a per-file 'un-setter' of this default?

**Comment by eddelbuettel** [comment]
Come to think about it, I could locally reset `options(warn)` before hitting the deprecated calls. But then I get dinged with a 'side-effects triggered'.  Less bad than a fail, I suppose...

**Comment by markvanderloo** [comment]
For packages we can probably get away with adding the option setting to `tests/tinytest.R`, so

```r
options(warn=1)
tinytest::test_package("foo")
```

I can add this to `setup_tinytest()` and ppl can do it explicitly if they run tests interactively. waddayathink?

**Comment by MichaelChirico** [comment]
+1 to this issue. Adding `options(warn=1)` improves the behavior of {tinytest} in this situation which I haven't yet fully understood:

https://github.com/eddelbuettel/nanotime/pull/156

Symptoms:

 - When the suite is passing, the uncaught warning is swallowed
 - When there is an error, _both_ the error _and_ the warning are bubbled up together

I haven't managed to make this into a reprex -- a simple file with `expect_true({ warning(); TRUE })` does show the warning, and ditto for `foo = \() { warning(); TRUE }; expect_true(foo())`.

That said, the behavior makes the dev experience harder because, when adding incremental code that introduces an error by mistake, we might suddenly be met with a flood of warnings _from code that we haven't touched_, which can be quite distracting.

**Comment by eddelbuettel** [comment]
I disagree with Michael here, and like what `tinytest` does in the package that I use it before.  As discussed here at length, those who desire to enable warning can already do so.

**Comment by MichaelChirico** [comment]
My point is this:

 - You are making your first contribution to pkgA
 - You write some nice code and helpfully add some tests
 - Agh, I made a mistake. The new test fails
 - What's this new warning? I didn't use any warnings in the code I wrote...
 - Now I am tilting at windmills. Unbeknownst to me, I _only_ have to fix my failing test and all will be well. Instead, I've started trying to figure out where the warning came from. The existing suite has 100s of tests, and I have to know the magic `options(warn=1)` incantation.

I think forcing test authors to be more explicit is almost always preferable. If you expect the warning, `expect_warning()`! This makes it 100% transparent to all users (including future self) that "nothing is awry here if you see a warning".

**Comment by eddelbuettel** [comment]
And indeed I did add `expect_warning()` in https://github.com/eddelbuettel/nanotime/commit/647dad574ebbe6a7a45c564601a176080714b04f as I mentioned to you in the other thread.
